label_studio_ml/examples/segment_anything_model/labelstudio.py

```diff
@@ -0,0 +1,205 @@
+        """
+        device = "cuda"     # if you're not using CUDA, use "cpu" instead .... good luck not burning your computer lol
+        sam = sam_model_registry["vit_h"](checkpoint="sam_vit_h_4b8939.pth")        # Note: YOU MUST HAVE THE MODEL SAVED IN THE SAME DIRECTORY AS YOUR BACKEND
+        sam.to(device=device)
+
+        predictor = SamPredictor(sam)
+        return predictor
+
+PREDICTOR = load_my_model()
+PREV_IMG_PATH = ""
+PREV_IMG = 0
+IMAGE_EMBEDDING = 0
+previous_id = 0
+
+# empty mask and indicator for no mask
+onnx_mask_input = np.zeros((1, 1, 256, 256), dtype=np.float32)
+onnx_has_mask_input = np.zeros(1, dtype=np.float32)
+
+onnx_model_path = "sam_onnx_quantized_example.onnx"
+ORT = onnxruntime.InferenceSession(onnx_model_path)
+
+class MyModel(LabelStudioMLBase):
+    def __init__(self, **kwargs):
+        super(MyModel, self).__init__(**kwargs)
+        from_name, schema = list(self.parsed_label_config.items())[0]
+        self.from_name = from_name
+        self.to_name = schema['to_name'][0]
+    
+    def predict(self, tasks, **kwargs):
+        """ Returns the predicted mask for a smart keypoint that has been placed."""
+        
+        # Use this to check times for your predictions
+        # print(f"Current data and time1: {str(datetime.datetime.now())}") # Current data and time1: 2023-04-16 18:56:09.361688 (ALMOST INSTANTANEOUS FROM THE RUN)
+
+        results = []
+        predictions = []
+        predictor = PREDICTOR
+
+        image_url = tasks[0]['data']['image']
+
+        print(f"Here are the kwargs now {kwargs}")
+
+        # getting the height and width of the image that you are annotating real-time 
+        height = kwargs['context']['result'][0]['original_height']
+        width = kwargs['context']['result'][0]['original_width']
+
+        # getting x and y coordinates of the keypoint
+        x = kwargs['context']['result'][0]['value']['x'] * width / 100
+        y = kwargs['context']['result'][0]['value']['y'] * height / 100
+
+        # label that you selected with the keypoint. If this is running into error, use the second line of code instead
+        label = kwargs['context']['result'][0]['value']['labels'][0]
+        keypointlabel = kwargs['context']['result'][0]['value']['keypointlabels'][0]
+
+
+        # loading the image you are annotating in local. The image_dir MUST MATCH THE DIRECTORY WHERE YOU IMPORTED YOUR ANNOTATIONS INTO LABEL STUDIO FROM
+        image_path = get_image_local_path(image_url, image_dir=os.getcwd())
+
+
+        # this is to speed up inference after the first time you selected an image
+        global PREV_IMG_PATH
+        global PREV_IMG
+        global IMAGE_EMBEDDING
+        global onnx_has_mask_input
+        global onnx_mask_input
+        global previous_id
+
+        if image_path != PREV_IMG_PATH:
+            PREV_IMG_PATH = image_path
+            split = image_path.split('-')[-1]
+            image = cv2.imread(f"./{split}")
```

Comment:
Why do you split image path?
It seems that it's already contain all the info to load the image

Reply:
Because it contains the ID for the image also, where if it was in a user's directory it would just be the name.

Ex.
Name of what gets returned from get_image_local_path  - _./f10bec-photo1.jpeg_ (something along these lines where after the last dash is whatever is in the user's local computer)

Name of the image in the user's directory - _./photo1.jpeg_

At least, this is how it appears when I run the code on my computer.

Reply:
Just went back and fixed this so there's no split and user doesn't have to worry about placing images in the right directory as long as its the one being annotated.